vcr/request.py

```diff
@@ -32,9 +33,19 @@ def __init__(self, method, uri, body, headers):
             self.body = body.read()
         else:
             self.body = body
-        self.headers = {}
-        for key in headers:
-            self.add_header(key, headers[key])
+        self.headers = CaseInsensitiveDict()
```

Comment:
I'd be happy to kill `add_header` (in fact I'd started to do that before I reined in the changes) but in that case it's not just the `headers` setter that needs changing, rather `headers` itself needs to have different behavior.

It would be like this (untested, just for discussion):
```python
class HeadersDict(CaseInsensitiveDict):
    def __setitem__(key, value):
        if isinstance(value, (tuple, list)):
            value = value[0]
        super(HeadersDict, self).__setitem__(key, value)
```

and then I'd move the giant docstring from `VCR` down to `HeadersDict`